### PR TITLE
#184 Converted classic pipeline to yaml - AWS Cloud Account Id Image Release 3.0 #193

### PR DESCRIPTION
**Pull request by priyadishah**
[https://github.com/https://github.com/https://github.com/https://github.com/robe070/cookbooks/issues/184]

Converted classic pipeline to YAML https://github.com/robe070/cookbooks/issues/184

Classic Pipeline
Repository: robe070/cookbooks
Branch: debug/paas
Release Pipeline Name - Cloud Account Id Image Release

Converted classic to yaml pipeline
Repository: priyadishah/cookbooks
Branch: feature/AWS-Cloud-Account-Id-Image-Release
Pipeline Name: AWS Publish Cloud Account Id Images 3.0
Pipeline Url: https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build?definitionId=67&_a=summary

### CODE REVIEW

File: AWS Publish Cloud Account Id Images.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      name: priyadishah/cookbooks
+      type: github
+      ref: feature/AWS-Cloud-Account-Id-Image-Release
+      endpoint: priyadishah
+
+pool:
+  vmImage: windows-2022
+
+stages:
+- template: templates/releaseTemplate.yml
```
Comment (robe070) [inline AWS Publish Cloud Account Id Images.yml]: You are using the wrong directory and wrong template name. I've provided the correct one. If you can't find them talk to Nikhil or Tarun for details

File: scripts/bake-scalable-azure-image-pipeline.ps1 [hunk 2]
```diff
@@ -56,7 +56,7 @@ param (
 
     [Parameter(Mandatory=$false)]
     [string]
-    $GitUserName="robe070",
+    $GitUserName="feature/aws-Build-Cloud-Account-Id-Artefacts-3.0",
```
Comment (robe070) [inline scripts/bake-scalable-azure-image-pipeline.ps1]: NO. Back out this change. And pass in the GitUserName from the YAML.

**Review by robe070 — CHANGES_REQUESTED** [review]

File: scripts/bake-scalable-azure-image-pipeline.ps1 (hunk 2 above)
Comment (priyadishah) [reply scripts/bake-scalable-azure-image-pipeline.ps1]: done

File: AWS Publish Cloud Account Id Images.yml (hunk 1 above)
Comment (priyadishah) [reply AWS Publish Cloud Account Id Images.yml]: @robe070  give me a day to make above changes, I was helping others for release pipelines so couldn't make changes.

**Comment by priyadishah** [comment]
Hi @robe070 any idea for this?https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2292&view=logs&j=25af95de-ccb4-5afc-7dbd-89130e10943a&t=e4a28591-4e13-54c1-8f44-1af86b74e3b7&l=33306

File: templates/releaseTemplate.yml [hunk 3]
```diff
@@ -54,11 +54,16 @@ stages:
...
+        path: "s/_lansa_aws-templates"
 
-      - checkout: _priyadishahCookbooks
-        displayName: Checkout cookbooks
+      # - checkout: _priyadishahCookbooks
+      #   displayName: Checkout cookbooks
+
+      - checkout: _robe070_cookbooks
+        displayName: Rob cookbooks
+        path: "s/_robe070_cookbooks"
```
Comment (robe070) [inline templates/releaseTemplate.yml]: Curious path "s/"?

File: templates/releaseTemplate.yml [hunk 4]
```diff
@@ -54,11 +54,16 @@ stages:
 
       - checkout: self
 
-      - checkout: _LansaAWSTemplates
-        displayName: Checkout aws-templates-md
+      - checkout: _lansa_aws-templates
+        displayName: Checkout _lansa_aws-templates
+        path: "s/_lansa_aws-templates"
```
Comment (robe070) [inline templates/releaseTemplate.yml]: Curious path "s/"?

File: AWS Publish Cloud Account Id Images.yml (hunk 1 above)
Comment (robe070) [reply AWS Publish Cloud Account Id Images.yml]: For close to 2 weeks?

**Comment by robe070** [comment]
> Hi @robe070 any idea for this?https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2292&view=logs&j=25af95de-ccb4-5afc-7dbd-89130e10943a&t=e4a28591-4e13-54c1-8f44-1af86b74e3b7&l=33306

I'm looking into this. Please do not run the Publish Cloud Account Id Images pipeline as it will delete the VMs that I'm looking at.

BUT, you have plenty of time to fix up the template names - just do not run it.

**Comment by priyadishah** [comment]
> > Hi @robe070 any idea for this?https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2292&view=logs&j=25af95de-ccb4-5afc-7dbd-89130e10943a&t=e4a28591-4e13-54c1-8f44-1af86b74e3b7&l=33306
> 
> I'm looking into this. Please do not run the Publish Cloud Account Id Images pipeline as it will delete the VMs that I'm looking at.
> 
> BUT, you have plenty of time to fix up the template names - just do not run it.

Sure @robe070 i have cancelled the job, please let me know if I need to do anything.

**Comment by robe070** [comment]
> fix up the template names

This is what you need to do now!

fix up the template names 

And I have not seen a canceled Run. Its just failed of its own accord. Please provide the link to the cancelled Run

**Comment by priyadishah** [comment]
> > fix up the template names
> 
> This is what you need to do now!
> 
> fix up the template names
> 
> And I have not seen a canceled Run. Its just failed of its own accord. Please provide the link to the cancelled Run

https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2294&view=results

ok, I am fixing up Template Name now,

**Comment by robe070** [comment]
> https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2294&view=results

I was talking about "Publish Cloud Account Id Images pipeline", not "Build Cloud Account Id Artefacts 3.0". 
You can run "Build Cloud Account Id Artefacts 3.0"

**Comment by robe070** [comment]
> > > fix up the template names
> > 
> > 
> > This is what you need to do now!
> > fix up the template names
> > And I have not seen a canceled Run. Its just failed of its own accord. Please provide the link to the cancelled Run
> 
> https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2294&view=results
> 
> ok, I am fixing up Template Name now,

When you have fixed it, just push the fix. Then I can check it. 
It should take you about 10 mins to do. So check it in NOW!
**Don't run it until I say its clear to do so.**

**Comment by priyadishah** [comment]
> > > > fix up the template names
> > > 
> > > 
> > > This is what you need to do now!
> > > fix up the template names
> > > And I have not seen a canceled Run. Its just failed of its own accord. Please provide the link to the cancelled Run
> > 
> > 
> > https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2294&view=results
> > ok, I am fixing up Template Name now,
> 
> When you have fixed it, just push the fix. Then I can check it. It should take you about 10 mins to do. So check it in NOW! **Don't run it until I say its clear to do so.**

just to clear I ran "Build Cloud Account Id Artefacts 3.0" pipeline.
image


I checked logs under "Publish Cloud Account Id Images 3.0" but why I am getting error 500, I am not sure. nothing found suspicious. and I am getting error 500 for only 1 URL.

**Comment by priyadishah** [comment]
> > > > > fix up the template names
> > > > 
> > > > 
> > > > This is what you need to do now!
> > > > fix up the template names
> > > > And I have not seen a canceled Run. Its just failed of its own accord. Please provide the link to the cancelled Run
> > > 
> > > 
> > > https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2294&view=results
> > > ok, I am fixing up Template Name now,
> > 
> > 
> > When you have fixed it, just push the fix. Then I can check it. It should take you about 10 mins to do. So check it in NOW! **Don't run it until I say its clear to do so.**
> 
> just to clear I ran "Build Cloud Account Id Artefacts 3.0" pipeline. image
> 
> I checked logs under "Publish Cloud Account Id Images 3.0" but why I am getting error 500, I am not sure. nothing found suspicious. and I am getting error 500 for only 1 URL.

Now, I am again checking "Publish Cloud Account Id Images 3.0" release pipeline. other than that, not doing anything until you update. @robe070

**Comment by robe070** [comment]
> I checked logs under "Publish Cloud Account Id Images 3.0" but why I am getting error 500, I am not sure. nothing found suspicious. and I am getting error 500 for only 1 URL.

I said I'm looking into this!

This PR is all about "Publish Cloud Account Id Images 3.0".

And because YOU sent me a link in this PR to "Build Cloud Account Id Artefacts 3.0" which YOU had cancelled, which is nothing to do with "Build Cloud Account Id Artefacts 3.0", I mentioned that you did NOT need to cancel it because thats not what I was talking about!

**Comment by robe070** [comment]
Please respond to my code questions above.

File: AWS Publish Cloud Account Id Images.yml (hunk 1 above)
Comment (priyadishah) [reply AWS Publish Cloud Account Id Images.yml]: i have fixed this and have pushed the code on my feature branch. 
image

File: templates/releaseTemplate.yml (hunk 3 above)
Comment (priyadishah) [reply templates/releaseTemplate.yml]: it was not picking correct path, worked with tarun he asked me add this

File: templates/releaseTemplate.yml (hunk 4 above)
Comment (priyadishah) [reply templates/releaseTemplate.yml]: it was not picking correct path, worked with tarun he asked me add this

**Comment by robe070** [comment]
I have resolved that the errors you are seeing in this pipeline are unimportant.

You should run with the default Variable settings that control which stages are run.

The important stages are not currently running because of a bug in the template you've created.

The stages not to run are:
image

The stages to run are:
image


The first task for the 3 stages that should be running should call:
```
SetGateVariable.ps1
```

Instead it calls
```
SetGateVariableCloudAccountId.ps1
```
The latter should be called for the stages that are not to be run currently.

As I've been saying for ages, fix the 3 that are not working. And you haven't.

**Comment by priyadishah** [comment]
> I have resolved that the errors you are seeing in this pipeline are unimportant.
> 
> You should run with the default Variable settings that control which stages are run.
> 
> The important stages are not currently running because of a bug in the template you've created.
> 
> The stages not to run are: image
> 
> The stages to run are: image
> 
> The first task for the 3 stages that should be running should call:
> 
> ```
> SetGateVariable.ps1
> ```
> 
> Instead it calls
> 
> ```
> SetGateVariableCloudAccountId.ps1
> ```
> 
> The latter should be called for the stages that are not to be run currently.
> 
> As I've been saying for ages, fix the 3 that are not working. And you haven't.

As you requested, I am using SetGateVariable.ps1 file for 3 stages and once those jobs run, we are getting as expected the response which we have set.

https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2319&view=logs&j=cab3b3e9-7299-5cff-680e-ccfc2562f2a7&t=45d5e4e6-c322-57c5-ed71-397cdf226324&l=25

**Comment by robe070** [comment]
You really need to look further into these issues. The log says what its looking for and cannot find:
`D:\a\1\s/_Build Image Release Artefacts/aws/w19d-15-0.txt`

and the earlier task which downloads the artefacts says this:
`D:\a\1/_Build Image Release Artefacts\aws/w19d-15-0.txt`

Spot the difference? (its not the '\' or '/' differences) I've already pointed out this issue to you.
cc: @nkhl-accolite

**Comment by priyadishah** [comment]
> You really need to look further into these issues. The log says what its looking for and cannot find: `D:\a\1\s/_Build Image Release Artefacts/aws/w19d-15-0.txt`
> 
> and the earlier task which downloads the artefacts says this: `D:\a\1/_Build Image Release Artefacts\aws/w19d-15-0.txt`
> 
> Spot the difference? (its not the '' or '/' differences) I've already pointed out this issue to you. cc: @nkhl-accolite

You are correct @robe070 but what's happening is

If I don't add "S" in git checkout, I am getting error below:
image


After adding "S" in git checkout, above error is gone but again path of "Download artifacts from Build Image Release Artefacts 3.0" & path of finding AMI are on different place. I tired to match path for mapping AMI to Download artifacts from Build Image Release Artefacts 3.0 with "system.defaultworkingdirectory" and  "pipeline.workspace" but couldn't make it.

"S" has been added only after confirmation with Tarun.

@priyadishah @nkhl-accolite

**Comment by robe070** [comment]
> If I don't add "S" in git checkout, I am getting error below: image
> 
> After adding "S" in git checkout, above error is gone but again path of "Download artifacts from Build Image Release Artefacts 3.0" & path of finding AMI are on different place. I tired to match path for mapping AMI to Download artifacts from Build Image Release Artefacts 3.0 with "system.defaultworkingdirectory" and "pipeline.workspace" but couldn't make it.
> 
> "S" has been added only after confirmation with Tarun.

This is an AzureDevOps issue. Its completely in your hands to resolve. And you have not provided me with what yml file changes you made to arrive at your conclusion above. So I can't comment even if I wanted to.

All I see is your latest yml file. It has this:
```
        - download: _Build Image Release Artefacts
          displayName: Download artifacts from Build Image Release Artefacts 3.0

        - checkout: self
        
        - checkout: _lansa_aws-templates
          displayName: Checkout Lansa AWS Templates
          path: "s/_lansa_aws-templates"

        - checkout: _robe070_cookbooks
          displayName: Checkout Rob cookbooks
          path: "s/_robe070_cookbooks"
```

Did you remove the 's' in these 2 paths? And if you did I suspect the remainder would be the default anyway.

The script is using `System.DefaultWorkingDirectory`. And that obviously contains an 's' 

Notice how the _Build Image Release Artefacts download doesn't have a path. What if you set it to have an 's'?

And why did I need to point this out?
@nkhl-accolite

File: templates/releaseTemplate.yml (hunk 3 above)
Comment (priyadishah) [reply templates/releaseTemplate.yml]: removed "S/" from path and it is resolved.

File: templates/releaseTemplate.yml (hunk 4 above)
Comment (priyadishah) [reply templates/releaseTemplate.yml]: removed "S/" from path and it is resolved.

**Comment by robe070** [comment]
Why didn't you use this solution?
_Notice how the _Build Image Release Artefacts download doesn't have a path. What if you set it to have an 's'?_

**Comment by priyadishah** [comment]
> Why didn't you use this solution? _Notice how the _Build Image Release Artefacts download doesn't have a path. What if you set it to have an 's'?_

I have discussed with Nikhil for this issue and after the discussion, Yes, I've removed the "s" because when there are multiple repos involved other than just the self, it's advisable to use path, and when you introduce path, the repo get's cloned into $(Pipeline.Workspace), which is D:/a/1

Besides, I read somewhere that MS is moving away from $(System.DefaultWorkingDirectly), and it will soon be replaced with
$(pipeline.workspace)

https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#properties:~:text=path%20string.,3.230.0%20and%20higher.

@nkhl-accolite @priyadishah

**Comment by robe070** [comment]
> > Why didn't you use this solution? _Notice how the _Build Image Release Artefacts download doesn't have a path. What if you set it to have an 's'?_
> 
> I have discussed with Nikhil for this issue and after the discussion, Yes, I've removed the "s" because when there are multiple repos involved other than just the self, it's advisable to use path, and when you introduce path, the repo get's cloned into $(Pipeline.Workspace), which is D:/a/1
> 
> Besides, I read somewhere that MS is moving away from $(System.DefaultWorkingDirectly), and it will soon be replaced with $(pipeline.workspace)
> 
> https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#properties:~:text=path%20string.,3.230.0%20and%20higher.
> 
> @nkhl-accolite @priyadishah

Thank you. Great for me to understand your reasoning. And I concur completely.

**Comment by priyadishah** [comment]
Hi @robe070 ,

Same error on AWS Cloud Account Id Image Release as well

https://dev.azure.com/VisualLansa/Lansa%20AWS%20Scalable%20License%20Images/_build/results?buildId=2512&view=logs&j=cab3b3e9-7299-5cff-680e-ccfc2562f2a7&t=b8359356-ed9f-5daf-31c2-caddc322a08b&l=227

**Pull request merged**